src/parser.rs

```diff
@@ -4378,6 +4397,50 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse `CREATE TABLE x AS TABLE y`
+    pub fn parse_as_table(&mut self) -> Result<Table, ParserError> {
+        let mut as_flag = false;
+        let mut table_flag = false;
+        let mut table_name = "";
+        let mut schema_name = "";
+
+        let mut tokens_iter = self.tokens.iter().enumerate().peekable();
```

Comment:
I find this somewhat confusing -- it parses the statement directly starting at the first token -- what if the first token was whitespace?

In fact, when I change the test to start with whitespace it failed:

```
#[test]
fn parse_create_table_as_table() {
    let sql1 = "   CREATE TABLE new_table AS TABLE old_table";
```

```
---- parse_create_table_as_table stdout ----
thread 'parse_create_table_as_table' panicked at 'called `Result::unwrap()` on an `Err` value: ParserError("Expected CREATE TABLE ... AS TABLE ..., found:  ")', src/test_utils.rs:86:61
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
```